apps/web/components/SideNav.tsx

```diff
@@ -298,6 +298,9 @@ export function SideNav(props: {
 function Sidebar(props: { isMobile: boolean }) {
   const path = usePathname();
   const showMailNav = path === "/mail" || path === "/compose";
+  const searchParams = useSearchParams();
+  // "?" is added to match the href strings of NavItems
+  const activePath = `?${showMailNav && searchParams.toString() ? searchParams.toString() : "type=inbox"}`; // if we are at "/mail" only, then the "Inbox" item will be highlighted
```

Comment:
Something like this would be cleaner and cover cases where there are other searchParams:

```ts
const params = new URLSearchParams(searchParams);
params.set(searchParam, value);
```

Reply:
This has also been changed for better readability and tested as well!